app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginFragment.java

```diff
@@ -0,0 +1,170 @@
+package com.youssef.fixit.Auth.LoginFragment;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.databinding.DataBindingUtil;
+import androidx.fragment.app.Fragment;
+
+import android.preference.PreferenceManager;
+import android.text.InputType;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.youssef.fixit.Models.Data.RetrofitClient;
+import com.youssef.fixit.Models.Bids.CreateBid;
+import com.youssef.fixit.Models.Register.Register;
+import com.youssef.fixit.R;
+import com.youssef.fixit.MainActivity.MainActivity;
+import com.youssef.fixit.databinding.FragmentLoginBinding;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class LoginFragment extends Fragment implements LoginView {
+    FragmentLoginBinding binding;
+    public SharedPreferences preferences;
+    public SharedPreferences.Editor editor;
+    public ProgressDialog dialog;
+    int showpassword;
+    LoginPresenter loginPresenter;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_login, container, false);
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        IntialSharedPreferences();
+        InitDialog();
+        InitViews();
+        ForgetPassword();
+        ShowPassword();
+    }
+
+    void InitViews() {
+        loginPresenter = new LoginPresenter(this);
+        binding.btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String Mail = binding.etMail.getText().toString();
+                String Password = binding.etPassword.getText().toString();
+                dialog.show();
```

Comment:
this one can be also sent to the presenter. In the presenter.onLogin first it will check if the data entered is valid or not then when it does an actual server request it will show loading
but no need to show loading before doing a long time process, what do you think?

Reply:
how to access dialog from the presenter

Reply:
you can create a new method in the view contract so the presenter will trigger view.showLoading() and in the activity you can show the dialog or any other loading option